src/rantanplan/structures.py

```diff
@@ -258,314 +178,320 @@ def is_terceto_encadenado(rhyme_pattern):
-        lambda lengths: statistics.median(lengths) == 8 or 9 > statistics.mean(lengths) > 7
+        lambda ranges_list: (
+                            has_same_length_verses(OCTOSYLLABLE, ranges_list)
+                        ) or (
+                            has_mixed_length_verses(OCTOSYLLABLE,
+                                                    TETRASYLLABLE, ranges_list))
     ), (
         CONSONANT_RHYME,
         "octava",
         r".{8}",
-        lambda lengths: all(23 > length > 3 for length in lengths)
+        lambda ranges_list: (
+                            has_minimum_length(ARTE_MAYOR_MIN_LENGTH, ranges_list)
+                        ) or (
+                            has_maximum_length(ARTE_MENOR_MAX_LENGTH, ranges_list))
     ), (
         ASSONANT_RHYME,
         "octava",
         r".{8}",
-        lambda lengths: all(23 > length > 3 for length in lengths)
+        lambda ranges_list: (
+                            has_minimum_length(ARTE_MAYOR_MIN_LENGTH, ranges_list)
+                        ) or (
+                            has_maximum_length(ARTE_MENOR_MAX_LENGTH, ranges_list))
     ), (
         CONSONANT_RHYME,
         "espinela",
         r"abbaaccddc",
-        lambda lengths: statistics.median(lengths) == 8 or 9 > statistics.mean(lengths) > 7
+        lambda ranges_list: has_same_length_verses(OCTOSYLLABLE, ranges_list)
     ), (
         CONSONANT_RHYME,
         "copla_real",
         r"""((ababa)|(abaab)|(abbab)|(aabab)|(aabba))
         ((ababa)|(abaab)|(abbab)|(aabab)|(aabba)
         (cdcdc)|(cdccd)|(cddcd)|(ccdcd)|(ccddc))""",
-        # tiene versos quebrados, cambiar regla de lengths
-        lambda lengths: statistics.median(lengths) == 8 or 9 > statistics.mean(lengths) > 7
+        # tiene versos quebrados, cambiar regla de ranges_list?
+        lambda ranges_list: has_same_length_verses(OCTOSYLLABLE, ranges_list)
     ), (
         CONSONANT_RHYME,
         "lira",
         r"ababb",
-        is_lira
+        lambda ranges_list: has_fixed_length_verses(STRUCTURES_LENGTH["lira"], ranges_list)
     ), (
         CONSONANT_RHYME,
         "quinteto",
         r"(ababa|abaab|abbab|aabab|aabba)",
-        lambda lengths: all(14 > length > 9 for length in lengths)
+        lambda ranges_list: has_minimum_length(ARTE_MAYOR_MIN_LENGTH, ranges_list)
     ), (
         CONSONANT_RHYME,
         "quintilla",
         r"(ababa|abaab|abbab|aabab|aabba)",
-        lambda lengths: all(11 > length > 3 for length in lengths)
+        lambda ranges_list: has_maximum_length(ARTE_MENOR_MAX_LENGTH, ranges_list)
     ), (
         ASSONANT_RHYME,
         "couplet",
         r"aa",
-        lambda lengths: all(20 > length > 1 for length in lengths)
+        lambda ranges_list: has_maximum_length(MAXIMUM_SAFE_LENGTH,
+                                               ranges_list) and (
+                                has_minimum_length(MINIMUM_SAFE_LENGTH, ranges_list))
```

Comment:
Maybe something like this?
```suggestion
        lambda ranges_list: (
            has_maximum_length(MAXIMUM_SAFE_LENGTH, ranges_list)
                and has_minimum_length(MINIMUM_SAFE_LENGTH, ranges_list))
```